### PR TITLE
scripts: partition_manager: Remove empty container partitions

### PR DESCRIPTION
**Pull request by oyvindronningstad**
If a container partition ends up empty after the first cleanup,
remove it.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

### CODE REVIEW

**Comment by oyvindronningstad** [comment]
cc @hakonfam @sigvartmh @osaether

**Comment by hakonfam** [comment]
Already some of the empty containers are deleted, could you elaborate which ones werent?

Also it would be great if there were unit tests which displayed what didn't work before, so that we won't get any regression, as this stuff is not easy to understand for a layman developer.

**Comment by hakonfam** [comment]
Note to self, this fixes the issue where partition manager complains when an app is built with B0 but not mcuboot.

File: scripts/partition_manager/partition_manager.rst [hunk 1]
```diff
@@ -94,8 +94,7 @@ span: list OR dict: list
    Partitions with this property are container partitions.
    Therefore, this property cannot be used together with the ``placement`` property.
 
-   The Partition Manager fails if a span is empty or if the partitions are not consecutive after processing.
-   Non-existing partitions are ignored.
+   Non-existing partitions are ignored removed from the ``span`` list before processing, and partitions with empty ``span`` lists are removed altogether (unless filled via ``inside``).
```
Comment (hakonfam) [inline scripts/partition_manager/partition_manager.rst]: `ignored removed`

Comment (oyvindronningstad) [reply scripts/partition_manager/partition_manager.rst]: fixed

**Comment by oyvindronningstad** [comment]
@b-gent This PR has one changed doc line. Want to have a look?

**Comment by b-gent** [comment]
Looking good 👍

**Pull request merged**